src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -60,6 +169,72 @@ impl ModulusPolynomialRingZq {
     }
 }
 
+#[cfg(test)]
+mod test_get_coeff_z {
+    use crate::{
+        integer::Z,
+        integer_mod_q::{ModulusPolynomialRingZq, Zq},
+        traits::GetCoefficient,
+    };
+    use std::str::FromStr;
+
+    /// Ensure that `0` is returned if the provided index is not yet set.
+    #[test]
+    fn index_out_of_range() {
+        let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+
+        let zero_coeff_1 = poly.get_coeff(4).unwrap();
+        let zero_coeff_2 = poly.get_coeff(4).unwrap();
+
+        assert_eq!(Z::ZERO, zero_coeff_1);
+        assert_eq!(Zq::from((0, 17)), zero_coeff_2);
+    }
+
+    /// Tests if coefficients are returned correctly.
+    #[test]
+    fn positive_coeff() {
+        let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+
+        let coeff_1 = poly.get_coeff(2).unwrap();
+        let coeff_2 = poly.get_coeff(2).unwrap();
+
+        assert_eq!(Z::from(2), coeff_1);
+        assert_eq!(Zq::from((2, 17)), coeff_2);
+    }
+
+    /// Tests if large coefficients are returned correctly.
+    #[test]
+    fn large_coeff() {
+        let poly =
+            ModulusPolynomialRingZq::from_str(&format!("2  1 {} mod {}", u64::MAX - 1, u64::MAX))
+                .unwrap();
+
+        assert_eq!(Z::from(u64::MAX - 1), poly.get_coeff(1).unwrap());
+        assert_eq!(
+            Zq::from((u64::MAX - 1, u64::MAX)),
+            poly.get_coeff(1).unwrap()
+        );
+    }
+
+    /// Tests if negative index yields an error in get_coeff with [`Z`].
+    #[test]
+    #[should_panic]
+    fn negative_index_error_z() {
+        let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+
+        let _: Z = poly.get_coeff(-1).unwrap();
+    }
+
+    /// Tests if negative index yields an error in get_coeff with [`Zq`].
+    #[test]
+    #[should_panic]
+    fn negative_index_error_zq() {
+        let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+
+        let _: Zq = poly.get_coeff(-1).unwrap();
+    }
+}
+
```

Comment:
please add a test for indexes that have not been set yet

Reply:
The first tests ensures this